source/plugins/ruby/KubernetesApiClient.rb

```diff
@@ -25,7 +25,12 @@ class KubernetesApiClient
   #@@IsValidRunningNode = nil
   #@@IsLinuxCluster = nil
   @@KubeSystemNamespace = "kube-system"
-  @LogPath = "/var/opt/microsoft/docker-cimprov/log/kubernetes_client_log.txt"
+  @os_type = ENV["OS_TYPE"]
+  if !@os_type.nil? && !@os_type.empty? && @os_type.strip.casecmp("windows") == 0
```

Comment:
wrapping this to helper function is good option to consider to avoid same code multiple places.

Reply:
Since this is across multiple files, I think wrapper function would be an overkill. Agree this would have helped if it was in the same file.

Reply:
helper function would be useful when that piece of code being used multiple places regardless whether thats same file or different file. it makes more readble and easy to manage as well.

Reply:
I think a better way to solve this problem is to keep it consistent between linux and windows. This is a workaround since we dont have the same path. When we moved to oneagent, I think one of the optimizations we can make is to use the same path.